python/cudf/cudf/_lib/copying.pyx

```diff
@@ -110,11 +108,15 @@ def gather(source, maps, bounds_check=True):
 
     for i, in_col in enumerate(in_cols):
         if isinstance(in_col, CategoricalColumn):
-            out_cols[i] = CategoricalColumn(
-                data=out_cols[i].data,
-                mask=out_cols[i].mask,
+            dtype = cudf.CategoricalDtype(
                 categories=in_col.cat().categories,
-                ordered=in_col.cat().ordered)
+                ordered=in_col.cat().ordered
+            )
+            out_cols[i] = column.build_column(
+                data=None,
+                mask=out_cols[i].mask,
+                dtype=dtype,
+                children=(out_cols[i],))
```

Comment:
This feels super unintuitive for building a CategoricalColumn

Reply:
OK I'll just replace all these calls to a special `build_categorical_column`

Reply:
Done